Bencodex.Tests/Types/DictionaryTest.cs

```diff
@@ -74,5 +74,144 @@ public void Equality()
             });
             Assert.NotEqual(a, c);
         }
+
+        [Fact]
+        public void SetItem()
+        {
+            var dictionary = Dictionary.Empty
+                .SetItem("text", "foo")
+                .SetItem("integer", 1337)
+                .SetItem("binary", new byte[] { 0x01, 0x02, 0x03, 0x04 })
+                .SetItem("boolean", true)
+                .SetItem("list", new IValue[] { (Text) "bar", (Integer) 1337 });
+
+            Assert.Equal("foo", (Text) dictionary["text"]);
+            Assert.Equal("foo", dictionary.GetValue<Text>("text"));
+            Assert.Throws<InvalidCastException>(
+                () => dictionary.GetValue<Integer>("text"));
+            Assert.Throws<InvalidCastException>(
+                () => dictionary.GetValue<Binary>("text"));
+            Assert.Throws<InvalidCastException>(
+                () => dictionary.GetValue<Bencodex.Types.Boolean>("text"));
+
+            Assert.Equal((Integer) 1337, (Integer) dictionary["integer"]);
+            Assert.Equal((Integer) 1337,
+                dictionary.GetValue<Integer>("integer"));
+            Assert.Throws<InvalidCastException>(
+                () => dictionary.GetValue<Binary>("integer"));
+            Assert.Throws<InvalidCastException>(
+                () => dictionary.GetValue<Text>("integer"));
+            Assert.Throws<InvalidCastException>(
+                () => dictionary.GetValue<Bencodex.Types.Boolean>("integer"));
+
+            Assert.Equal(
+                new byte[] { 0x01, 0x02, 0x03, 0x04 },
+                (Binary) dictionary["binary"]);
+            Assert.Equal(
+                new byte[] { 0x01, 0x02, 0x03, 0x04 },
+                dictionary.GetValue<Binary>("binary"));
+            Assert.Throws<InvalidCastException>(
+                () => dictionary.GetValue<Integer>("binary"));
+            Assert.Throws<InvalidCastException>(
+                () => dictionary.GetValue<Text>("binary"));
+            Assert.Throws<InvalidCastException>(
+                () => dictionary.GetValue<Bencodex.Types.Boolean>("binary"));
+
+            Assert.Equal(
+                (Bencodex.Types.Boolean) true,
+                (Bencodex.Types.Boolean) dictionary["boolean"]);
+            Assert.Equal(
+                (Bencodex.Types.Boolean) true,
+                dictionary.GetValue<Bencodex.Types.Boolean>("boolean"));
+            Assert.Throws<InvalidCastException>(
+                () => dictionary.GetValue<Integer>("boolean"));
+            Assert.Throws<InvalidCastException>(
+                () => dictionary.GetValue<Binary>("boolean"));
+            Assert.Throws<InvalidCastException>(
+                () => dictionary.GetValue<Text>("boolean"));
+
+            Assert.Equal(
+                new IValue[] { (Text)"bar", (Integer)1337 },
```

Comment:
Code style seems inconsistent: `(T) v` vs `(T)v`, but it's okay, we could introduce lints & style checkers in the separate pull request.

Reply:
I replaced them like `(T) v`, to maintain consistent already exists in this project.
After this pull-request is merged, I will open the pull-request you mentioned.